### DISCUSSION TITLE
Active-Active Replication - How many DCs?

### DISCUSSION THREAD

**Discussion by effectivelywild** [minio/minio#16451] [Q&A]
In the active replication FAQ section it reads:

_At present, MinIO only recommends replication across two data centers. It is possible to have replication across multiple data centers, however, the complexity involved and the tradeoffs required make this rather difficult._

But in other sections of the Active-Active replication it makes no such mention of this recommendation and shows examples that list more than two DCs.

_Starting with our latest [MinIO server release](https://github.com/minio/minio/releases/tag/RELEASE.2021-09-23T04-46-24Z), MinIO supports Multi-Site, Active-Active replication for synchronization of objects between an arbitrary number of MinIO deployments._

https://blog.min.io/minio-multi-site-active-active-replication/

What's the actual best practice here?

**Comment by harshavardhana** [comment]
// cc @ravindk89 this needs to be updated I believe.

**Reply by ravindk89** [reply]
So:

- https://min.io/product/active-data-replication-for-object-storage <- source of the originally quoted paragraph
- https://blog.min.io/minio-multi-site-active-active-replication/ - 2021 blog on bucket replication only
- https://min.io/docs/minio/linux/operations/install-deploy-manage/multi-site-replication.html - docs for site replication

We can revisit the best practices here and document them more clearly, along with linking the older blog to more current information.

@fmuise We have since implemented site replication, which has reduced operational complexity insofar as setting up initial replication between peer sites or adding new peer. that said, the guidance on our main web page is accurate. The more sites you have (with site replication) or buckets you have (with bucket replication), the more complexity you have in management. 

That's secondary to a 'best practice' though. Really our general guideline here is to understand your own infrastructure, it's capabilities, and it's limitations. If your infra between data centers has high latency, low availability, spotty connection, etc. - replication will not work the way you (or we) want it to. But if you have good infra between data centers - low latench, high availability, strong stable connections - then the number of sites or peers becomes less of a concern. From there you deal with hardware (are the actual backing nodes of your MinIO sites sized appropriately for the workload?) and similar concerns.

Beyond those general guidelines, we have SUBNET - because the testing and validation process tends to be fact-specific, and is something best handled as part of professional engagement.
